### ISSUE TITLE
WARNING: tensorflow: Model failed to serialize as JSON. Ignoring... Layer YoloLayer has arguments in `__init__` and therefore must override `get_config`.

### ISSUE THREAD

**Issue by marbortoli** [OlafenwaMoses/ImageAI#706]
Hello everyone,
I'm trying to run a custom model detection as the example script with the hololens

_"from imageai.Detection.Custom import DetectionModelTrainer
trainer = DetectionModelTrainer()
trainer.setModelTypeAsYOLOv3()
trainer.setDataDirectory (data_directory = "/content/hololens")
trainer.setTrainConfig (object_names_array = ["hololens"], batch_size = 4, num_experiments = 100, train_from_pretrained_model = "/content/pretrained-yolov3.h5")
trainer.trainModel()"_

however, I always get this error message just after the first epochs:
_WARNING:tensorflow:Model failed to serialize as JSON. Ignoring... Layer YoloLayer has arguments in `__init__` and therefore must override `get_config`._
...
_CustomMaskWarning: Custom mask layers require a config and must override get_config. When loading, the custom mask layer must be passed to the custom_objects argument.
  warnings.warn('Custom mask layers require a config and must override '_

I don't understand how to fix it. Can someone help me?
I'm running on tensorflow 2.7, VScode, CPU 

Thanks a lot :)

**Comment by LucienMP** [comment]
Is this warning important and how did you fix it? @marbortoli

**Comment by marbortoli** [comment]
hey! I was having issues running the code on my local computer.. so it was important to me,

I followed this issue here:
https://stackoverflow.com/questions/61390089/ignoring-layers-with-arguments-in-init-must-override-get-config

I added the def get_config.... in my yolov3 script
:)

**Comment by gdubbs100** [comment]
I'm having this issue too - It comes about when I run the simple training example (although I'm using a different dataset).

I think this issue - the warning regarding failing to serialize a json - and the issue with model evaluation (also resulting from running the object detection evaluation tutorial with the specified libraries) where it evaluates everything as zero might be related?

Based on the error messages alone, it looks like the model configuration might not be saved correctly, and then the evaluation cannot read it correctly.

**Comment by farizqinanda** [comment]
@marbortoli which yolov3 script did you add with def get_config? Is it at /usr/local/lib/python3.7/dist-packages/imageai/Detection/YOLO/yolov3.py?

**Comment by 347115683** [comment]
i hvae the same issue

**Comment by 347115683** [comment]
I have the same issue
